public/css/icinga/forms.less

```diff
@@ -1,618 +1,231 @@
 
-.control-button,
-input[type="submit"] {
-  .button();
+.form-element,
+.form-controls {
+  display: flex;
 }
 
-input:-moz-placeholder { // FF 18-
-  color: @gray-light;
-  opacity: 1;
-}
+.form-control {
+  display: inline-block;
+  float: right;
 
-input::-moz-placeholder { // FF 19+
-  color: @gray-light;
-  opacity: 1;
+  &:not(:last-child) {
+    margin-left: .5em;
+  }
+
+  &:last-child {
+    .clearfix();
+  }
 }
 
-input:-ms-input-placeholder {
-  color: @gray-light;
+.control-group {
+  display: flex;
 }
 
-input::-webkit-input-placeholder {
-  color: @gray-light;
+/**********************************************************************************************************************
+                                                 label styles
+***********************************************************************************************************************/
+
+.control-group .control-label-group {
+  display: inline-block;
+  //todo(jem): make text-align depend on width of form (when flex switches to one-col the labels don't align
+  margin: .5em 1em 0.5em 0;
+  overflow: hidden;
+  text-align: right;
+  text-overflow: ellipsis;
+  vertical-align: baseline;
+  white-space: nowrap;
+  width: 14em;
+  //todo(jem): maybe solve width with flexbox flex-growth: 1
+}
+
+/**********************************************************************************************************************
+                                               general input styles
+***********************************************************************************************************************/
+
+//todo(jem): solve width with flex-growth: 6 combined with label
+.control-group input[type="text"],
+.control-group input[type="password"],
+.control-group textarea,
+.control-group select {
+  background: @low-sat-blue;
+  flex-grow: 1;
+}
+
+.control-group .control-label-group,
+.control-group input,
+.control-group button,
+.control-group select,
+.control-group textarea,
+.form-controls button,
+.form-controls input {
+  border: none;
+  border-radius: .25em;
+  font-size: inherit;
+  margin: .5em 1em 0.5em 0;
+  padding: .5625em;
+  .appearance(none);
 }
 
-input.search {
-  background: transparent url("../img/icons/search.png") no-repeat scroll 0.1em center;
-  background-size: 1em 1em;
-  padding-left: 1.5em;
+/**********************************************************************************************************************
+                                             specific input styles
+***********************************************************************************************************************/
 
-  &:focus {
-    background-color: @body-bg-color;
-    background-image: url('../img/icons/search_icinga_blue.png') !important;
-  }
+.control-group textarea {
+  background-image: url(../static/img?file=textarea-corner-2x.png&module_name=reporting);
```

Comment:
The URL is wrong.

Reply:
fixed in c2b886598579c3293727954afff8f8e383cdd73e